src/shared_types.rs

```diff
@@ -177,7 +178,12 @@ impl Serialize for Guideline {
         let (x, y, angle) = match self.line {
             Line::Vertical(x) => (Some(x), None, None),
             Line::Horizontal(y) => (None, Some(y), None),
-            Line::Angle { x, y, degrees } => (Some(x), Some(y), Some(degrees)),
+            Line::Angle { x, y, degrees } => {
+                if !(0.0..=360.0).contains(&degrees) {
+                    return Err(ser::Error::custom("angle must be between 0 and 360 degrees."));
```

Comment:
as with `Identifier`, my preference here would be to make illegal values impossible to construct. This could be follow-up work, though.

Reply:
That was my thinking. At which level though? making a special `Degrees` newtype?